PowerEditor/src/dpiManagerV2.cpp

```diff
@@ -54,72 +107,62 @@ void DPIManagerV2::initDpiAPI()
 		HMODULE hUser32 = ::GetModuleHandleW(L"user32.dll");
 		if (hUser32 != nullptr)
 		{
-			ptrFn(hUser32, _fnGetDpiForSystem, "GetDpiForSystem");
-			ptrFn(hUser32, _fnGetDpiForWindow, "GetDpiForWindow");
-			ptrFn(hUser32, _fnGetSystemMetricsForDpi, "GetSystemMetricsForDpi");
-			ptrFn(hUser32, _fnSystemParametersInfoForDpi, "SystemParametersInfoForDpi");
-			ptrFn(hUser32, _fnSetThreadDpiAwarenessContext, "SetThreadDpiAwarenessContext");
-			ptrFn(hUser32, _fnAdjustWindowRectExForDpi, "AdjustWindowRectExForDpi");
+			LoadFn(hUser32, _fnGetDpiForSystem, "GetDpiForSystem");
+			LoadFn(hUser32, _fnGetDpiForWindow, "GetDpiForWindow");
+			LoadFn(hUser32, _fnGetSystemMetricsForDpi, "GetSystemMetricsForDpi");
+			LoadFn(hUser32, _fnSystemParametersInfoForDpi, "SystemParametersInfoForDpi");
+			LoadFn(hUser32, _fnIsValidDpiAwarenessContext, "IsValidDpiAwarenessContext");
+			LoadFn(hUser32, _fnSetThreadDpiAwarenessContext, "SetThreadDpiAwarenessContext");
+			LoadFn(hUser32, _fnAdjustWindowRectExForDpi, "AdjustWindowRectExForDpi");
 
 		}
 	}
 }
 
 int DPIManagerV2::getSystemMetricsForDpi(int nIndex, UINT dpi)
 {
-	if (_fnGetSystemMetricsForDpi != nullptr)
-	{
-		return _fnGetSystemMetricsForDpi(nIndex, dpi);
-	}
-	return DPIManagerV2::scale(::GetSystemMetrics(nIndex), dpi);
+	return _fnGetSystemMetricsForDpi(nIndex, dpi);
+}
+
+BOOL DPIManagerV2::isValidDpiAwarenessContext(DPI_AWARENESS_CONTEXT value)
+{
+	return _fnIsValidDpiAwarenessContext(value);
 }
 
 DPI_AWARENESS_CONTEXT DPIManagerV2::setThreadDpiAwarenessContext(DPI_AWARENESS_CONTEXT dpiContext)
 {
-	if (_fnSetThreadDpiAwarenessContext != nullptr)
+	return _fnSetThreadDpiAwarenessContext(dpiContext);
+}
+
+DPI_AWARENESS_CONTEXT DPIManagerV2::setThreadDpiAwarenessContextSafe(DPI_AWARENESS_CONTEXT dpiContext)
+{
+	if (DPIManagerV2::isValidDpiAwarenessContext(dpiContext) == TRUE)
 	{
 		return _fnSetThreadDpiAwarenessContext(dpiContext);
 	}
-	return NULL;
+	return nullptr;
```

Comment:
setThreadDpiAwarenessContext can be removed.
BTW, the new setThreadDpiAwarenessContextSafe() is safer, but makes quite confusion IMO.